All sort and dataframe related changes from my branch

SteveNY-Tibco opened the pull request.

SteveNY-Tibco committed:
Making columns of DataFrame rdered

Since sort operation sorts both by column (axis = 0) and by row (axis = 1), we need a way to represent the order of row. This modification only for making DataFrame keep the order of its columns.

SteveNY-Tibco committed:
Adding sort operation

SteveNY-Tibco committed:
Add for sort operation

SteveNY-Tibco committed:
changes about DataFrame and sort operation

1. Finalize sort operation.
2. Enhance DataFrame, DataFrameSorter, SortableTuple
3. modifies operation according to the change of DataFrame

SteveNY-Tibco committed:
Fix some bugs and better test case

SteveNY-Tibco committed:
remove KeepRow parameter from sort

abramvandergeest commented:
Generally looks really good.  My one major comment is that the dataframe type is for internal use only it (currently) cannot be an input or output of an operation.  (Previously since Dataframe was a strait map this wasn't really a problem, but now it is since we don't define a df in the specification.)  I am totally open with discussing this during the meeting tomorrow.
Minor comments:
in sort/operation.go:
       Lines 43-45 should be in the New function (line 23) (default params values are usually set there)
       L57-62 should be debug not info (saying we entered/exited an operation is info, printing inputs/outputs is debug.

SteveNY-Tibco committed:
Update common.go

1. Add setting source data structure flag
2. Remove output order for map type of data

SteveNY-Tibco committed:
Update sort operator

documentation & descriptor

SteveNY-Tibco committed:
Update groupBy

1. Using new DataFrame stuff
2. Better printing in test
3. documentation and descriptor

SteveNY-Tibco committed:
Update trasnpose

1. Using new DataFrame stuff
2. Better printing in test
3. documentation and descriptor

SteveNY-Tibco committed:
Update join

documentation and descriptor

SteveNY-Tibco committed:
Update pivot

1. Better printing in test
2. documentation and descriptor

SteveNY-Tibco committed:
Update join

Better printing in test

SteveNY-Tibco committed:
Update logging

SteveNY-Tibco committed:
Merge branch 'master' into master
